### ISSUE TITLE
TR1 Bugs : Bug report after completing the game

### ISSUE THREAD

**Issue by lemmi23** [XProger/OpenLara#183]
Hello XProger,

I made a full run of tomb raider 1 with OpenLara. It was amazing to get a chance to play this game I haven't played back since I was a kid. **Thank you for this amazing work !!** I had a blast :)

My run went by like a breeze. I did encounter a few bugs that I decided to list here. There are 2 bugs that could be a real problem, but there is some workaround to get to past them. **So, the game is (almost) finishable on a normal run !!! :)**

I played on windows 10, mainly with the .exe from the last release (January 11). Then I replayed the webGL versions and the recompiled .exe from the github sources (from the 30 of april) to see if the bugs where still there on both versions and took notes. So the bugs listed there have been encountered at least 3 times.

Here is the list (sorry if my english isn't great, it's not my mother language) :

### MAJORS BUGS: 
_(that could stop players from finishing the game/access secret, on a normal playthrough)_

- **Grab Slope**
Location : Level 8C
This one has already been reported here : https://github.com/XProger/OpenLara/issues/120
This is the only time I was really stuck on my progression (Luckily, The magic "o" key saved the day !)

- **Boulder puzzle doesn't reset**
Location : Level 10B / Boulder trap and block puzzle
Bug : The TRAP_BOULDER (127) doesn't reset when exiting the room, making the puzzle impossible to finish. Luckily, since the boulders can be passthrough, it doesn't prevent you from finishing the game (I just walk through the boulder to get to the next room)
![TR1-boulder_puzzle](https://user-images.githubusercontent.com/50141823/57141102-c0a61d00-6db9-11e9-88b1-be91a072e083.png)

- **Boulder traps doesn't work**
Location : Level 10A / Boulder traps room
Bug : The TRAP_BOULDER (33) doesn't work properly. The boulder never get down to the end of the slope. It also doesn't reset when exiting the room.
![TR1-boulder_trap_33](https://user-images.githubusercontent.com/50141823/57141111-c7349480-6db9-11e9-927d-0036a893b51e.jpg)

- **Ledge to secret :**
Location : Level 10C
Bug : Those were already reported there : https://github.com/XProger/OpenLara/issues/123

- **Can't get on the building**
Location : Level 10A / hanging building
Bug : Lara can reach the ledge of the hanging building once it is on the floor, but she can't climb up on it. It's not stopping progression from the main path but prevent from getting to a secret.
![TR1-building_10A](https://user-images.githubusercontent.com/50141823/57141137-d4518380-6db9-11e9-9103-9acccbce50b7.jpg)

- **Bullseye Jump**
Location : Level 10C / next to the third secret
Bug : Right before fighting with Natla, there is a jump into a tiny square of water. It seems like high jumps don't work the same way as the original game (at least, on this particular jump). It feel like Lara get momentum from jumping, making her land far off-course and making this last jump really hard to get right. Here is how it work in the original release (https://youtu.be/LtkMeyRpwCQ?t=21963) and how it works when doing the same jump in Open Lara.
![TR1-bulleyes_jump_gif](https://user-images.githubusercontent.com/50141823/57141148-da476480-6db9-11e9-81c6-f8b12a674660.gif)
![TR1-bulleyes_jump](https://user-images.githubusercontent.com/50141823/57141255-1975b580-6dba-11e9-87b9-4e344ede89c0.png)

### MINOR BUGS

- **Ledge in front of the Sphynx**
Location : Level 8C / in front of the Sphynx
Bug : This ledge didn't go as expected, see the gif (speed x3).
![TR1-ledge_8C_01](https://user-images.githubusercontent.com/50141823/57141169-ea5f4400-6db9-11e9-98cb-6b519b23bf00.jpg)
![TR1-ledge_8C_02](https://user-images.githubusercontent.com/50141823/57141181-f21ee880-6db9-11e9-8d64-c78e0d163ddf.gif)

- **Stuck enemies**
Location : All levels
Bug : Enemies get often stuck when they have to go trough a door. Especially gorillas, lions, mummies and mutants. It's not much of an issue in Peru, but get more problematic in some levels in Greece and Egypt since there is a lot rooms with smaller doors.
![TR1-ennemies_stuck](https://user-images.githubusercontent.com/50141823/57141201-fcd97d80-6db9-11e9-8ecf-69698b4ff61c.gif)
![TR1-mummy_bug](https://user-images.githubusercontent.com/50141823/57141213-0367f500-6dba-11e9-8ac5-035c905265fc.gif)

- **Last FMV don't play**
Location : End of the game
Bug : When I finished the game, after the big slope, the game when straight to the main title menu without playing the ending. I tested with FMV files, both inside "FMV" folder or inside "video/1/" folder. The other FMV worked fine. _I don't know if the problem is still there with RPL files, though._

- **"Invisible" Human Enemies**
Location : All levels with human enemies
Bug : Happen often after an encounter with a human enemy. After they "left" the level, it triggers Lara hurt animation (without affecting health level) at the place the enemy where when they last "disappear". As if they where still there but invisible.
![TR1-invisible-hurt](https://user-images.githubusercontent.com/50141823/57141217-06fb7c00-6dba-11e9-8b8d-956404cf27ba.gif)

- **Fire sound fx**
Location : All levels
Bug : The sound design of Lara bursting into flames is a lot louder than the rest of SFX.

- **Getting trough boulders**
Location : All Levels
Bug : Lara can get trough boulders
![TR1-through-boulders](https://user-images.githubusercontent.com/50141823/57141226-0bc03000-6dba-11e9-802d-de7166c140d3.PNG)

- **Getting through mutants "eggs"**
Location : Level 10B / Level 10C
Bug : Lara can get through the demon eggs at the start of the Atlantid.
![TR1-through-mutants-eggs](https://user-images.githubusercontent.com/50141823/57141237-124ea780-6dba-11e9-8e35-ebfb0fe2e5b5.png)

- **Swimming Rats**
Location : Level 6 / Fire Trial Room
Bug : Rats in the water are not swimming at the surface but are at the bottom
![TR1-swiming_rats](https://user-images.githubusercontent.com/50141823/57141280-22ff1d80-6dba-11e9-8c78-f33437ae6022.png)

I had a few other bugs like the skateboard kid without his skateboard or the hanging building being doubled (hanging and crashed to the ground at the same time). But I couldn't replicate those bugs in the WebGL version. So I assume you fixed those issues or there are not encounter that often.

I hope this post will be helpful. Thank you so much for this great work and sharing it with everyone :)

Cheers !

Charles

**Comment by johnnyt83** [comment]
I would like to add that the "lead bars" in the Midas Palace level are too dark. It took me almost an hour to find the last lead bar on top of the temple because it was almost invisible. It is way brighter in the original versions. Would be great if this is fixed.

**Comment by lemmi23** [comment]
Yes, I had that issue as well. And also with one of the fuse in Natla's mine.
But, if you turn on the "simple object" option (it's on the graphic option), it make all the items 2D sprites and they are easier to see. I'm pretty sure that how they looked in the original PS version.

**Comment by johnnyt83** [comment]
Good to know. Will try that.
